### ISSUE TITLE
[BUG] setup.cfg attr: value is inconsistent between parsing and loading module

### ISSUE THREAD

**Issue by jasongrout** [pypa/setuptools#2602]
### setuptools version

setuptools===49.6.0.post20210108

### Python version

Python 3.9.2

### OS

macOS Catalina

### Description

In https://github.com/pypa/setuptools/blob/b2f7b8f92725c63b164d5776f85e67cc560def4e/setuptools/config.py#L35 StaticModule parses the statements in order to find a relevant assignment, but it should parse statements in *reverse* order to be consistent with the behavior when loading a module (i.e., the last assignment wins when loading a module).

### Expected behavior

I expect the parsing shortcut for finding module attributes to be consistent with loading the module.

### How to Reproduce

This inconsistency is illustrated when having a project with a setup.cfg that contains `version = attr: mypackage.__version__`, where `mypackage/__init__.py` is

```python
__version__ = "0.1.0"
__version__ = "0.4.0"
```
With this setup, the version will be 0.1.0, but if the first line is changed to something `literal_parse` does not understand, forcing setuptools to load the module (like `__version__ = "0.1.0" + ""`), then the version is 0.4.0.

### Output

```console

```

### Code of Conduct

- [X] I agree to follow the PSF Code of Conduct

**Comment by telamonian** [comment]
@jasongrout You might want to widen the snippet of problematic code:

https://github.com/pypa/setuptools/blob/23ee037d56a6d8ab957882e1a041f67924ae04da/setuptools/config.py#L33-L39

Initially I was confused, since it seemed like forward iterating over all of the literal assignment statements corresponding to an `attr` would indeed produce the correct result. The key thing here is that `next` causes only at most the first value of the generator comprehension to ever be evaluated

**Comment by pganssle** [comment]
I agree that this is better, but presumably another option here is to throw an exception if the same value is defined more than once?

I think we don't want to get too clever with `attr:`, as that way lies madness, and I can't think of a reason someone would do this except by accident in a simple module?
